slim/train_image_classifier.py

```diff
@@ -343,19 +345,11 @@ def _get_init_fn():
 
   exclusions = []
   if FLAGS.checkpoint_exclude_scopes:
-    exclusions = [scope.strip()
+    exclusions = [re.escape(scope.strip())
                   for scope in FLAGS.checkpoint_exclude_scopes.split(',')]
 
-  # TODO(sguada) variables.filter_variables()
-  variables_to_restore = []
-  for var in slim.get_model_variables():
-    excluded = False
-    for exclusion in exclusions:
-      if var.op.name.startswith(exclusion):
-        excluded = True
-        break
-    if not excluded:
-      variables_to_restore.append(var)
+  var_list = slim.get_model_variables()
+  variables_to_restore = slim.filter_variables(var_list=var_list, exclude_patterns=exclusions, reg_search=False)
```

Comment:
On a little closer inspection of the API, I could probably use `slim.get_variables_to_restore(exclude=exclusions)` and avoid the call to `slim.get_model_variables()` as well as wrapping the exclusions with `re.escape`